src/scripts/cronjobs/emailBreachAlerts.tsx

```diff
@@ -411,8 +404,22 @@ async function init() {
   });
   await initEmail();
 
-  const [subClient, receivedMessages] = await pullMessages();
-  await poll(subClient, receivedMessages ?? []);
+  let messageCount = 0
+  const [subClient, formattedSubscription] = createPubSubClient();
+  while (maxMessages === 0 || messageCount < maxMessages) {
+    // If there are no messages, this will wait until the default timeout for the pull API.
+    // @see https://cloud.google.com/pubsub/docs/pull
+    console.debug("polling pubsub...");
+    const [response] = await subClient.pull({
+      subscription: formattedSubscription,
+      maxMessages: 10,
+    });
+    const receivedMessages = response.receivedMessages ?? []
+    messageCount = messageCount + receivedMessages.length
+    console.debug(`pulled ${receivedMessages.length} messages`)
+
+    await poll(subClient, receivedMessages);
```

Comment:
If the subscription is empty, the loop will keep calling `subClient.pull` in rapid succession?  I wonder if it's best to add a short delay (e.g., `await new Promise(res => setTimeout(res, 1000))`) between polls when zero messages are returned, to avoid hammering the API and wasting CPU cycles.

Reply:
The `subClient.pull()` method will wait for messages in the queue for a period of time (can't find any documentation on how long) but probably on the order of 10s of seconds